ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/api/P2PNetwork.java

```diff
@@ -77,21 +77,23 @@
   void subscribeDisconnect(DisconnectCallback consumer);
 
   /**
-   * Adds a {@link Peer} to a list indicating efforts should be made to always stay connected to it
+   * Adds a {@link Peer} to a list indicating efforts should be made to always stay connected
+   * regardless of maxPeer limits. Non-permitted peers may be added to this list, but will not
+   * actually be connected to as long as they are prohibited.
    *
    * @param peer The peer that should be connected to
-   * @return boolean representing whether or not the peer has been added to the list or was already
-   *     on it
+   * @return boolean representing whether or not the peer has been added to the list, false is
+   *     returned if the peer was already on the list
    */
   boolean addMaintainConnectionPeer(final Peer peer);
```

Comment:
Previously, peer permission would be checked as peers were added to this list.  The changes in this PR delay the point at which we're able to check permission to the point where our local node is up and running.  As a result, if we were to gate entrance into the list based on permissions, there would be a period where no nodes could be added to this list while the node is starting up.  So, this API has been updated such that a peer can be added to the "maintain connection" list whether or not it is permitted on the network.  However, no peers will actually be connected to without first checking permissions.